src/WebJobs.Script/Workers/SharedMemoryDataTransfer/MemoryMappedFileAccessor.cs

```diff
@@ -52,5 +53,73 @@ protected void ValidatePlatform(OSPlatform platform)
                 throw new PlatformNotSupportedException("Cannot instantiate on this platform");
             }
         }
+
+        /// <summary>
+        /// Write the given <see cref="bool"/> value into the <see cref="MemoryMappedViewStream"/>.
+        /// </summary>
+        /// <param name="mmv"><see cref="MemoryMappedViewStream"/> to write to.</param>
+        /// <param name="value">The <see cref="bool"/> value to write.</param>
+        private static void WriteBool(MemoryMappedViewStream mmv, bool value)
+        {
+            // The boolBytes array will be of length 1 so we pick the first byte
+            byte[] boolBytes = BitConverter.GetBytes(value);
+            byte boolByte = boolBytes[0];
+            mmv.WriteByte(boolByte);
+        }
+
+        /// <summary>
+        /// Read a <see cref="bool"/> value from the <see cref="MemoryMappedViewStream"/>.
+        /// </summary>
+        /// <param name="mmv"><see cref="MemoryMappedViewStream"/> to read from.</param>
+        /// <returns>The read <see cref="bool"/> value.</returns>
+        private static bool ReadBool(MemoryMappedViewStream mmv)
+        {
+            int boolInt = mmv.ReadByte();
+            // ReadByte will return -1 if the byte could not be read
+            if (boolInt == -1)
+            {
+                return false;
+            }
+            byte boolByte = (byte)boolInt;
+            byte[] boolBytes = { boolByte };
+            bool flag = BitConverter.ToBoolean(boolBytes, 0);
+            return flag;
+        }
+
+        /// <summary>
+        /// Checks if the dirty bit of the <see cref="MemoryMappedFile"/> has been set or not.
+        /// If it is set then it means the <see cref="MemoryMappedFile"/> was created and may already be in use.
+        /// If it is not set then this <see cref="MemoryMappedFile"/> is new and can be used.
+        /// </summary>
+        /// <param name="mmf">The <see cref="MemoryMappedFile"/> to check.</param>
+        /// <returns><see cref="true"/> if the <see cref="MemoryMappedFile"/> is new and does not contain any content, <see cref="false"/> otherwise.</returns>
+        protected bool IsDirtyBitSet(MemoryMappedFile mmf)
```

Comment:
I think dirty bit has implications in caches, we probably should use IsMmapBitSet or something that reflects better.